hibernate-reactive-core/src/test/java/org/hibernate/reactive/UpdateOrderingTestBase.java

```diff
@@ -0,0 +1,307 @@
+
+	public static class UpdateOrderingTrueTest extends UpdateOrderingTestBase {
+
+		@Override
+		public boolean isUpdateOrdering() {
+			return true;
+		}
+
+		@Override
+		public String getBatchSize() {
+			return "0";
+		}
+
+		@Override
+		public void assertLoggedQueries() {
+			assertThat( sqlTracker.getLoggedQueries() ).hasSize( 30 );
+			assertThat( sqlTracker.getLoggedQueries().get( 0 ) ).startsWith( "update Category set name=" );
+			assertThat( sqlTracker.getLoggedQueries().get( 1 ) ).startsWith( "update Category set name=" );
+			assertThat( sqlTracker.getLoggedQueries().get( 14 ) ).startsWith( "update Category set name=" );
+			assertThat( sqlTracker.getLoggedQueries().get( 15 ) ).startsWith( "update Product set name=" );
+			assertThat( sqlTracker.getLoggedQueries().get( 16 ) ).startsWith( "update Product set name=" );
+			assertThat( sqlTracker.getLoggedQueries().get( 29 ) ).startsWith( "update Product set name=" );
+			sqlTracker.clear();
+		}
+	}
+
+	public static class UpdateOrderingFalseTest extends UpdateOrderingTestBase {
+
+		@Override
+		public boolean isUpdateOrdering() {
+			return false;
+		}
+
+		@Override
+		public String getBatchSize() {
+			return "0";
+		}
+
+		@Override
+		public void assertLoggedQueries() {
+			assertThat( sqlTracker.getLoggedQueries() ).hasSize( 30 );
+			assertThat( sqlTracker.getLoggedQueries().get( 0 ) ).startsWith( "update Category set name=" );
+			assertThat( sqlTracker.getLoggedQueries().get( 1 ) ).startsWith( "update Product set name=" );
+			sqlTracker.clear();
+		}
+	}
+
+	//	Test that when we set the Batch size only the expected number of queries run.
+	//	For example, setting hibernate.jdbc.batch_size to 50 will only log two update queries in this case (when order_inserts is true)
+	public static class UpdateOrderingBatchSizeTest extends UpdateOrderingTestBase {
+
+		@Override
+		public boolean isUpdateOrdering() {
+			return true;
+		}
+
+		@Override
+		public String getBatchSize() {
+			return "50";
+		}
+
+		@Override
+		public void assertLoggedQueries() {
+			assertThat( sqlTracker.getLoggedQueries() ).hasSize( 2 );
+			assertThat( sqlTracker.getLoggedQueries().get( 0 ) ).startsWith( "update Category set name=" );
+			assertThat( sqlTracker.getLoggedQueries().get( 1 ) ).startsWith( "update Product set name=" );
+			sqlTracker.clear();
+		}
+	}
+
+	/**
+	 * @return value of property {@link Settings#ORDER_UPDATES}
+	 */
+	public abstract boolean isUpdateOrdering();
+
+	/**
+	 * run checks expected logged queries via {@link SqlStatementTracker#getLoggedQueries()}
+	 */
+	public abstract void assertLoggedQueries();
+
+	/**
+	 * @return value of property {@link Settings#STATEMENT_BATCH_SIZE}
+	 */
+	public abstract String getBatchSize();
+
+	public SqlStatementTracker sqlTracker;
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.setProperty( Settings.ORDER_INSERTS, "false" );
```

Comment:
Could you also test the inserts, please? We do the inserts in the `populatedb`,  so the test only needs to check the content of `sqlTracker`.

Reply:
done